swift/apple/FirezoneKit/Sources/FirezoneKit/Views/AppView.swift

```diff
@@ -41,10 +41,23 @@ public class AppViewModel: ObservableObject {
         self.status = status
 
 #if os(macOS)
-        if status == .invalid || DeviceMetadata.firstTime() {
-          AppViewModel.WindowDefinition.main.openWindow()
-        } else {
-          AppViewModel.WindowDefinition.main.window()?.close()
+        Task {
+          var firezoneId: FirezoneId?
+
+          do {
+            try await firezoneId = FirezoneId.load()
+          } catch {
+            // no-op; firezoneId remains nil if it can't be loaded
+          }
+
+          if status == .invalid || firezoneId == nil {
```

Comment:
This nil check is redundant unless you wanted this code to run within a `finally` block? You won't get here if there is no ID because it doesn't throw an exception.

Reply:
It falls through - have cleaned this up so it should be a little more apparent.